### ISSUE TITLE
make gnome-shell extensions compatible

### ISSUE THREAD

**Issue by cyberorg** [linuxmint/Cinnamon#103]
Here is the workflow for getting gnome-shell extensions(most) working with cinnamon for now:
- copy gnome-shell extensions to ~/.local/share/cinnamon/extensions/
- Run the following commands:
  cd ~/.local/share/cinnamon/extensions/
  for i in `ls`; do
  sed -i -e "s@imports.gi.Shell@imports.gi.Cinnamon@g" $i/*
  grep -q cinnamon-version $i/metadata.json || sed -i -e '/{/{n;n;n;s/\,/\,\n"cinnamon-version": [ "1.1.3" ],/;}' $i/metadata.json
  done

Attached patch makes browser plugin work partially, i.e. https://extensions.gnome.org/local/ works fully, but installation of plugins does not work. I do not know any programming but getting the above workflow automated via browser plugin should be fairly simple for someone who knows how to code :)

In short, when user selects the extension to install on extensions.gnome.org, the plugin should download  it and make the above changes automatically.

http://paste.opensuse.org/93726837

**Comment by thorsummoner** [comment]
Was there movement on compatibility with gnome extensions? I don't know how different the cinnamon and gnome capabilities are in theory, but surely most plugins could be supported with a breif translation layer at most.

**Comment by rik-shaw** [comment]
Also wondering about this.  Where can I find a reference guide?
